climada/hazard/trop_cyclone.py

```diff
@@ -740,50 +743,31 @@ def _compute_windfields_sparse(
 
     # convert track variables to SI units
     si_track = tctrack_to_si(track, metric=metric)
-    t_lat, t_lon = si_track["lat"].values, si_track["lon"].values
-
-    # normalize longitudinal coordinates of centroids
-    u_coord.lon_normalize(coastal_centr[:, 1], center=si_track.attrs["mid_lon"])
-
-    # Restrict to the bounding box of the whole track first (this can already reduce the number of
-    # centroids that are considered by a factor larger than 30).
-    max_dist_eye_lat = max_dist_eye_km / u_const.ONE_LAT_KM
-    max_dist_eye_lon = max_dist_eye_km / (
-        u_const.ONE_LAT_KM * np.cos(np.radians(np.abs(coastal_centr[:, 0]) + max_dist_eye_lat))
-    )
-    coastal_idx = coastal_idx[
-        (t_lat.min() - coastal_centr[:, 0] <= max_dist_eye_lat)
-        & (coastal_centr[:, 0] - t_lat.max() <= max_dist_eye_lat)
-        & (t_lon.min() - coastal_centr[:, 1] <= max_dist_eye_lon)
-        & (coastal_centr[:, 1] - t_lon.max() <= max_dist_eye_lon)
-    ]
-    coastal_centr = centroids.coord[coastal_idx]
 
-    # After the previous filtering step, finding and storing the reachable centroids is not a
-    # memory bottle neck and can be done before chunking.
-    track_centr_msk = get_close_centroids(
-        t_lat, t_lon, coastal_centr, max_dist_eye_km, metric=metric,
+    # when done properly, finding and storing the close centroids is not a memory bottle neck and
+    # can be done before chunking:
+    centroids_close, mask_close, mask_close_alongtrack = get_close_centroids(
+        si_track, centroids.coord[idx_centr_filter], max_dist_eye_km, metric=metric,
     )
```

Comment:
Why would this function return both the close centroids and the mask for the close centroids? I see no reason why the "closeness" information should be carried around twice. What am I missing?

Reply:
The lon-coordinates of the returned centroids are normalized. We could of course apply the mask, and then normalize outside of the function (again). But that was the approach that caused the original bug that is fixed with this PR: We normalized in several places, and then in one place we forgot to normalize and everything silently "failed". 

On the other hand, we cannot just return the centroids without the mask, because other parts of the code need to know which centroids from the original array are the ones that we mask out.

Reply:
Indeed, I missed the fact that centroids are 1D, but the masks are 2D. Very good! Thanks for the clarification both here and in the code.